### PR TITLE
Add variable number of discriminator updates for each generator update

### LINKED ISSUE #101: Discriminator should train more than generator
The CTGAN does not implement the extra update iterations of the critic, when compared to the original implementation of WGAN and WGAN-GP. Is this intentional? In my experiments, I have seen that sometimes the generator becomes too powerful and exploits the discriminator. I think this might be improved by implementing the critic extra updates, compared to the generator. 

image
From the paper, we see the critic is updated 5 times more compared to the generator.

### PR DESCRIPTION
**Pull request by Baukebrenninkmeijer**
As per WGAN, the discriminator should be updated 5 times more often than the generator. This is added as a parameter in the `__init__` function. 

Fixes #101

### CODE REVIEW

File: ctgan/synthesizer.py [hunk 1]
```diff
@@ -32,10 +32,13 @@ class CTGANSynthesizer(object):
             Wheight Decay for the Adam Optimizer. Defaults to 1e-6.
         batch_size (int):
             Number of data samples to process in each step.
+        n_discriminator (int):
+            Number of discriminator updates to do for each generator update.
+            Defaults to 5.
```
Comment (fealho) [inline ctgan/synthesizer.py:37]: I'm guessing you meant ```Defaults to 1```. Everything else looks good to me :thumbsup:

Comment (Baukebrenninkmeijer) [reply ctgan/synthesizer.py:37]: Oh, correct. Forgot to change that one.

Comment (Baukebrenninkmeijer) [reply ctgan/synthesizer.py]: I added the WGAN default value in the docstring and the actual default in the normal place.

**Review by csala — CHANGES_REQUESTED** [review]
Thanks @Baukebrenninkmeijer It looks good!

I would make a couple of minor changes that I think will improve the overall result:

1. I would rename the parameter to something like `discriminator_steps`. Otherwise, at first sight someone may thing that `n_discriminators=5` means "Use 5 different discriminators".

2. Providing some more context in the docstring would be helpful. I would add a sentence to explain what is the effect of increasing or decreasing the value, a link to the WGAN paper, and then mention that the default is 1 to match the original CTGAN implementation.

Would you mind adding these? Once it is done I think we are all set!

**Comment by Baukebrenninkmeijer** [comment]
@csala Done and done

**Review by csala — APPROVED** [review]
Looks good now! Thanks @Baukebrenninkmeijer !

**Comment by Baukebrenninkmeijer** [comment]
Some checks are not clearing due to unrelated requirements issues it seems. 

@csala @fealho

**Comment by csala** [comment]
> Some checks are not clearing due to unrelated requirements issues it seems.

Yes, sometimes there are random errors related to the Github Actions infrastructure. I'm re-running the tests.

**Comment by Baukebrenninkmeijer** [comment]
Hmmm a value count in the samples was too high. But that seems also unrelated with my changes. Is that bound maybe too tight? Its 6550isch when it should be below 6500.

**Pull request merged**